StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c

```diff
@@ -311,7 +311,10 @@ MmIplCopyGuidHob (
 
   UsedSize = 0;
   GuidHob  = GetFirstGuidHob (Guid);
-  ASSERT (GuidHob != NULL);
+
+  if (GuidHob == NULL) {
+    DEBUG ((DEBUG_WARN, "GUIDed HOB %g could not be found\n", Guid));
+  }

```

Comment:
We should not just replace the assertion with a debug log.
Some HOBs should exist, and the assertion is to catch the missing HOBs.
If you feel some HOBs are optional, you can let MmIplCopyGuidHob return something, and the caller can choose to print a debug log or assert.